src/db.c

```diff
@@ -428,6 +429,7 @@ long long emptyDbStructure(redisDb *dbarray, int dbnum, int async,
  * DB number is out of range, and errno is set to EINVAL. */
 long long emptyDb(int dbnum, int flags, void(callback)(dict*)) {
     int async = (flags & EMPTYDB_ASYNC);
+    int with_functions = !(flags & EMPTYDB_NOFUNCTIONS);
```

Comment:
Probably should update the above comment to indicate "extra" data like functions may also be flushed. Right now it just says all keys are removed, which is what I would expect this function to do since "functions" aren't in databases. Maybe we should rename emptyDb to `emptyData` To make that distinction clear?

Reply:
Thanks @madolson , fixed.